docs/contribute/refactoring-filesystem.md

```diff
@@ -0,0 +1,105 @@
+A Saros filesystem consist of resources, called `IResource`. It contains its `IReferencePoint` and can either be an `IFolder` or an `IFile`.
+
+## Concepts how to reach the final state
+
+The refactoring of the project-based architecture to a reference point based one is very complicated. Especially the relationsships between the components needs a long analysis to find an effictive strategy to refactor Saros.
+An important strategy for the refactory is to build the reference point based architecture, but also keeping the 
+project-based architrecture simultaneously until the refactoring is completed.
+For providing the dual based architecture a special mapper is needed, which maps the reference point to the IDE specific file system resources. Because every components has different resources, a special mapper (ReferencePointManager) is requiered for each  of them.
+
+### Saros Core
+Saros Core needs a mapper, which is called CoreReferencePointManager, with following features:
+
+1.  It maps a reference points object to a Saros project object internally 
+2. It provides the functionalities of the Saros project
+
+Because a reference point itself has no methods, it is important that the CoreReferencePointManager provides functionalities of a Saros project, so that co-existance of both architecture is possible.
+
+
+### Saros/E
+
+In situations, like processing activities, Saros/E converts Saros resources backs to Eclipse resource. So it is very useful, that the mapper in Saros/E gets the features, that it determinates Eclipse resources given by reference points and relative paths to the resources. This mapper is called EclipseReferencePointManager and has following features:
+
+1.  It maps a reference point object to an Eclipse project internally
+2.  It determines Eclipse resources in combination of reference points and relative paths to the resources
+
+### Saros/I
+
+The IntelliJReferencePointManager is needed for Saros/I and has two responsibilities:
+
+1.  It maps a reference point object to an IntelliJ Module internally
+2.  It determines java resources in combination of reference points and relative paths to the resources
+
+### Saros Server
+
+The ServerReferencePointManager is needed for Saros Server and has two responsibilities:
+
+1.  It maps a reference point object to an Java IO file resource internally
+2.  It determines java resources in combination of reference points and relative paths to the resources
+
+All ReferencePointManager are not needed only during the refactoring. There are also needed especially handling Saros activities (ex. A saros resources was created). All activities contains a Saros path (SPath), which consists of a Saros project and a relative path to the resource, which the activity triggered. The Spath has additionally feature: It also returns the affected Saros resources.
```

Comment:
```suggestion
All ReferencePointManager's are also needed after the refactoring. They are useful especially for handling Saros activities (e.g. A saros resource was created). All activities contain a Saros path (SPath), which consists of a Saros project and a relative path to the resource, which the activity triggered. The SPath has additional features: It also returns the affected Saros resources.
```
Do you want to describe why the ReferencePointManagers are still useful in the context of the SPath? I don't really understand what you want to express. Is the last sentence "It also returns the affected Saros resources" a reason for using the ReferencePointManager?

Reply:
I wanted to explain why the ReferencePointManagers are needed. 

1. During the refactoring for providing the co-existence of the referencepoint- and project-based architectures.
2. After the refactoring (but also during the refactoring): It's not possible to determine resources (IDE-dependent or Saros resources), if you only have a reference point and a relative path. Because the ReferencePointManager maps a reference point to a (container) resource (like Eclipse project or IntelliJ module), you can get the mapped object and can determine the resource with the relative path.

The SPath is a special case: The current situation is, that the SPath contains a Saros project object and a IPath object as relative path. A SPath instance can return contained the Saros project and IPath object, but can also determine and return resources in the combination Saros project and IPath as relative path. After the refactoring, the SPath will contain a reference point and an IPath. Determining resources without a ReferencePointManager is not possible, so the functions like getFolder(), getFile() in SPath wont work anymore and will be removed.
But if you want to get the resource from the SPath, the ReferencePointManager can be used for this.

Reply:
@m273d15  Should I rewrite this or is this explanation enough?